pallets/parachain-staking/src/lib.rs

```diff
@@ -2233,10 +2429,11 @@ pub mod pallet {
 			collator_id: T::AccountId,
 			amt: BalanceOf<T>,
 		) -> Weight {
-			if let Ok(amount_transferred) = T::Currency::deposit_into_existing(&collator_id, amt) {
+			// Mint rewards to the collator
+			if let Ok(minted) = T::Currency::mint_into(&collator_id, amt) {
```

Comment:
This code seems redundant, we should call `mint(amt, collator_id)`

Reply:
Amended https://github.com/moonbeam-foundation/moonbeam/pull/3306/commits/e0d05698cbd8ba6efe95ebf9578eaca7ff7f0693